File: VICSApp/CoreSystem/DHCD/FormUI/Vote/Matter_ins_update.vb
Public Class Matter_ins_update
    Dim controlcode As String
    Dim updatemattercode As Integer
    Public Sub New(ByVal controlcode As String, ByVal mattercode As Integer)
        Me.controlcode = controlcode
        Me.updatemattercode = mattercode
        ' This call is required by the Windows Form Designer.
        InitializeComponent()
        ' Add any initialization after the InitializeComponent() call.

    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        Me.Close()
    End Sub

    Private Sub Matter_ins_update_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        MaskedTextBox1.Text = Mainform.workingmeeting
        If controlcode = "Update" Then
            NumericUpDown1.Value = Me.updatemattercode
            NumericUpDown1.ReadOnly = True
            Button1.Text = "Cập nhật"
            Me.Text = "Cập nhật vấn đề biểu quyết"
            Dim dt As New DataTable
            Try
                dt = Mainform.BenlyDal.Matter_getlist(Mainform.workingmeeting, updatemattercode)
            Catch ex As Exception
                MsgBox("lỗi" + ex.Message)
                Exit Sub
            End Try
            MaskedTextBox3.Text = dt.Rows(0).Item("Mattername")
            TextBox1.Text = dt.Rows(0).Item("MatterDescription")
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If controlcode = "Add" Then
            Try
                Mainform.BenlyDal.Matter_insert(Mainform.workingmeeting, NumericUpDown1.Value, MaskedTextBox3.Text, TextBox1.Text)
                Me.Close()
            Catch ex As Exception
                MsgBox("Lỗi : " + ex.Message)
            End Try
        ElseIf controlcode = "Update" Then
            Try
                Mainform.BenlyDal.Matter_update(Mainform.workingmeeting, NumericUpDown1.Value, MaskedTextBox3.Text, TextBox1.Text)
                Me.Close()
            Catch ex As Exception
                MsgBox("Lỗi : " + ex.Message)
            End Try

        End If
    End Sub
End Class

File: VICSApp/CoreSystem/DHCD/FormUI/Vote/Matter_ins_update.cs
using System;
using System.Data;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace pmDHCD
{
    public partial class Matter_ins_update
    {
        private string controlcode;
        private int updatemattercode;

        public Matter_ins_update(string controlcode, int mattercode)
        {
            this.controlcode = controlcode;
            updatemattercode = mattercode;
            // This call is required by the Windows Form Designer.
            InitializeComponent();
            _Button3.Name = "Button3";
            _Button1.Name = "Button1";
            // Add any initialization after the InitializeComponent() call.

        }

        private void Button3_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void Matter_ins_update_Load(object sender, EventArgs e)
        {
            MaskedTextBox1.Text = My.MyProject.Forms.Mainform.workingmeeting;
            if (controlcode == "Update")
            {
                NumericUpDown1.Value = updatemattercode;
                NumericUpDown1.ReadOnly = true;
                Button1.Text = "Cập nhật";
                Text = "Cập nhật vấn đề biểu quyết";
                var dt = new DataTable();
                try
                {
                    dt = My.MyProject.Forms.Mainform.BenlyDal.Matter_getlist(My.MyProject.Forms.Mainform.workingmeeting, updatemattercode);
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("lỗi" + ex.Message);
                    return;
                }

                MaskedTextBox3.Text = Conversions.ToString(dt.Rows[0]["Mattername"]);
                TextBox1.Text = Conversions.ToString(dt.Rows[0]["MatterDescription"]);
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (controlcode == "Add")
            {
                try
                {
                    My.MyProject.Forms.Mainform.BenlyDal.Matter_insert(My.MyProject.Forms.Mainform.workingmeeting, NumericUpDown1.Value.ToString(), MaskedTextBox3.Text, TextBox1.Text);
                    Close();
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi : " + ex.Message);
                }
            }
            else if (controlcode == "Update")
            {
                try
                {
                    My.MyProject.Forms.Mainform.BenlyDal.Matter_update(My.MyProject.Forms.Mainform.workingmeeting, NumericUpDown1.Value.ToString(), MaskedTextBox3.Text, TextBox1.Text);
                    Close();
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi : " + ex.Message);
                }
            }
        }
    }
}